lib/tools.js

```diff
@@ -2247,14 +2247,15 @@ function maybeCallback(callback, ...args) {
  * @returns {Promise<any>} - if Promise is resolved with multiple arguments, an array is returned
  */
 function maybeCallbackWithError(callback, error, ...args) {
+    const isDbError = (error instanceof Error ? error.message : (error || '').toString()) === ERROR_DB_CLOSED;
```

Comment:
This should be below the following `if`, so you don't have to convert the error again.

Reply:
yes, also makes the check much easier ... you are right